staging/src/k8s.io/apiserver/pkg/cel/library/format.go

```diff
@@ -0,0 +1,131 @@
+package library
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation"
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// Format provides a CEL library exposing common named Kubernetes string
+// validations.
+//
+// named(name: string) -> ?format
+//
+//	Returns the format with the given name, if it exists. Otherwise, optional.none
+//	Allowed names are: "dns1123label", "dns1123subdomain", "dns1035label", "qualifiedName"
+//
+// validate(format: format, str: string) -> ?list<string>
+//
+//	Validates the given string against the given format. Returns optional.none
+//	if the string is valid, otherwise a list of validation error strings.
+//
+// validatePrefix(format: format, str: string) -> ?list<string>
+//
+//	Same as validate, but assumes the input string will be used as a prefix for
+//	a kubernetes generated name.
```

Comment:
Im not sure that masking would be used for all prefixes. It is used for the dns related ones for sure. (I am also not sure if it is even correct, since the DNS validations check maxLength, but the mask does not append a full generated name)

Reply:
Good ponit, maybe https://github.com/kubernetes/kubernetes/pull/123572#discussion_r1508001485 is better here.